### PR TITLE
OCLOMRS-17: Protect routes that require authentication

### PR DESCRIPTION
**Pull request by emasys**
# JIRA TICKET NAME:
[OCLOMRS-17: Protected access of urls](https://issues.openmrs.org/browse/OCLOMRS-17)

# Summary:
After a user logs in, the other urls or routes need to be private to ensure authentication and confidentiality.

### CODE REVIEW

File: src/ActionCreators/AuthActionCreators.js [hunk 1]
```diff
@@ -13,4 +11,16 @@ const LoginFailed = errorMessage => ({
   payload: { errorMessage },
 });
 
-export { AUTHENTICATED, AUTHENTICATION_FAILED, Login, LoginFailed };
+const isAuthenticated = () => {
+  let payload = false;
+  const token = localStorage.getItem('openmrs-token');
```
Comment (hadijahkyampeire) [inline src/ActionCreators/AuthActionCreators.js]: Just wonder why you name your token openmrs-token, did you look at the name used when setting it in the localstorage

Comment (hadijahkyampeire) [reply src/ActionCreators/AuthActionCreators.js]: LGTM

**Comment by mwaz** [comment]
LGTM

**Comment by dkayiwa** [comment]
Changes in this pull request seem to be dealing with more than the ticket description.

File: src/Actions/AuthActions.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-      if (error.response) {
-        dispatch(LoginFailed(error.response.data.detail));
-      } else if (error.request) {
-        dispatch(LoginFailed("Request can't be made"));
-      }
+const LoginAction = ({ username, token }) => async (dispatch) => {
+  try {
+    const url = `${BASE_URL}/users/${username}/`;
+    const headers = { Authorization: `Token ${token}` };
+    // dispatch login sent
```
[the remark this answers is not in the document]
Comment (emasys) [reply src/Actions/AuthActions.js]: Not really, it wasn't part of the changes I made to this file though.

**Comment by dkayiwa** [comment]
Some merge conflicts here

**Pull request merged**